### DISCUSSION TITLE
How to prevent unstability of whole nx mess for simple fullstack app?

### DISCUSSION THREAD

**Discussion by igortas** [nrwl/nx#22922] [General]
Hi,

I'm trying to learn nx and keep it clean as possible, without adding new mess from the current one is generated by ```nx``` plugins.
I'm trying to create dead simple fullstack app, but everytime, I stop writing some code, turn off my pc and return back to the project, trying to continue like normal, sane person,  run the same command for the apps in parallel, ```nx``` behave, like I've uninstalled ```windows 11``` and switched to ```linux```.

Any tips, how ```nx``` can stay not broken, for at least half hour?

For example, I've return back to the nx demo app, and for some reason ```jest``` throws weird error, that never happend. Funny is, I need to run only the apps, not tests, the error is:

```
Failed to process project graph.
  The "@nx/jest/plugin" plugin threw an error while creating nodes from apps/api/jest.config.ts:
    Error: ● Validation Error:
```

Yesterday, I was facing cache issues forcing me to manually remove the ```.nx``` cache, so I can start the project.

Installing or upgrading the packages, nx gets frozen with only red error: ```NX lock files changed```, once again remove the ```.nx```` cache manually.

Two days before, I've replaced ```vite.config.ts``` file to ```vite.config.mts``` and ```nx``` keeps pointing to ```.mts.[hash].mjs``` file, once again, removing manually the cache.

``run-many``` not working sequentially, prevents from runing the next app(s), after initial one (in my case nodejs api)

I can continue going on, out of nowhere blockers.

If anyone has suggestions more how to proceed using nx, without this issues, I like to hear.

BR,

**Comment by Boushell** [comment]
Nx be whack as hell sometimes. 

What I did was rename it jest.config.ts.bak instead of deleting. 

Whatever it's doing checking this file can be bypassed by deleting or renaming. Who needs tests

**Comment by jaysoo** [comment]
Hi @igortas We're aware of some stability issues around Nx plugins. Can you try setting `NX_ISOLATE_PLUGINS=true` when running Nx? You can add it to your `.env`, for example. This option will allow plugins to parse configuration files in isolation, as to not pollute anything in the global scope. Loading some config files have unexpected side-effects, so this option helps to solve those issues.

Also checkout the latest Nx 19 release, which has other stability fixes.

If you have further issues, I would recommend opening an issue and a repro so we can investigate further.
